src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -54,19 +57,52 @@ void ColumnBetweenTableScanImpl::_scan_generic_segment(const BaseSegment& segmen
 
     auto typed_left_value = type_cast_variant<ColumnDataType>(_left_value);
     auto typed_right_value = type_cast_variant<ColumnDataType>(_right_value);
-    auto comparator = [typed_left_value, typed_right_value](const auto& position) {
-      return position.value() >= typed_left_value && position.value() <= typed_right_value;
-    };
 
-    _scan_with_iterators<true>(comparator, it, end, chunk_id, matches);
+    if (_left_inclusive && _right_inclusive) {
```

Comment:
The first step is done, renaming Between to BetweenInclusive and it still compiles.
Btw, regarding the parameter naming in with_comparator, why is it called 'func' instead of 'comparator' as in the method name?